### PR TITLE
feat(excel2xml): add support for 3-digit-years in find_date_in_string() (DEV-3005)

### PR DESCRIPTION
**Pull request by jnussbaum**

### CODE REVIEW

File: test/unittests/commands/excel2xml/test_excel2xml_lib.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def test_make_xsd_id_compatible(self) -> None:
         Path("special-characters.xml").unlink()
 
     def test_find_date_in_string(self) -> None:
-        # template: 2021-01-01 | 2015_01_02
+        # template: 2021-01-01
```
Comment (jnussbaum) [inline test/unittests/commands/excel2xml/test_excel2xml_lib.py:191]: some of the inline comments were in the wrong place, so I fixed them

File: src/dsp_tools/commands/excel2xml/excel2xml_lib.py [hunk 2]
```diff
@@ -134,7 +135,7 @@ def find_date_in_string(string: str) -> Optional[str]:
     startyear: Optional[int] = None
     endyear: Optional[int] = None
 
-    year_regex = r"([0-2][0-9][0-9][0-9])"
+    year_regex = r"([0-2]?[0-9][0-9][0-9])"
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/commands/excel2xml/excel2xml_lib.py:138]: Minus years are not supported right?

File: src/dsp_tools/commands/excel2xml/excel2xml_lib.py [hunk 3]
```diff
@@ -161,7 +162,7 @@ def find_date_in_string(string: str) -> Optional[str]:
     monthname_date_regex = rf"{lookbehind}({all_months}) ?{day_regex}, ?{year_regex}{lookahead}"
     monthname_date = regex.search(monthname_date_regex, string)
     # template: 1849/50 | 1849-50 | 1849/1850
-    year_range = regex.search(lookbehind + year_regex + r"[/-](\d{2}|\d{4})" + lookahead, string)
+    year_range = regex.search(lookbehind + year_regex + r"[/-](\d{2,4})" + lookahead, string)
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/commands/excel2xml/excel2xml_lib.py:165]: If you are working on that, could you not also build in:
1640-1
as in 1640-1641

File: src/dsp_tools/commands/excel2xml/excel2xml_lib.py (hunk 2 above)
Comment (jnussbaum) [reply src/dsp_tools/commands/excel2xml/excel2xml_lib.py:138]: No, that would be a bit more difficult

**Review by Nora-Olivia-Ammann — APPROVED** [review]

File: src/dsp_tools/commands/excel2xml/excel2xml_lib.py (hunk 3 above)
Comment (jnussbaum) [reply src/dsp_tools/commands/excel2xml/excel2xml_lib.py]: Done: https://github.com/dasch-swiss/dsp-tools/pull/652/commits/9311344f35e03c11a13f4463e47ae02c7f589320

**Review by BalduinLandolt — APPROVED** [review]

**Pull request merged**